src/sinks/hdfs/config.rs

```diff
@@ -0,0 +1,190 @@
+use codecs::{encoding::Framer, JsonSerializerConfig, NewlineDelimitedEncoderConfig};
+use opendal::{layers::LoggingLayer, services::Hdfs, Operator};
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, DataType, Input},
+    sink::VectorSink,
+};
+
+use crate::{
+    codecs::{Encoder, EncodingConfigWithFraming, SinkType},
+    config::{GenerateConfig, SinkConfig, SinkContext},
+    sinks::{
+        opendal_common::*,
+        util::{
+            partitioner::KeyPartitioner, BatchConfig, BulkSizeBasedDefaultBatchSettings,
+            Compression,
+        },
+        Healthcheck,
+    },
+};
+
+/// Configuration for the `hdfs` sink.
+///
+/// The Hadoop Distributed File System (HDFS) is a distributed file system
+/// designed to run on commodity hardware. HDFS consists of a namenode and a
+/// datanode. We will send rpc to namenode to know which datanode to send
+/// and receive data to. Also, HDFS will rebalance data across the cluster
+/// to make sure each file has enough redundancy.
+///
+/// ```txt
+///                     ┌───────────────┐
+///                     │  Data Node 2  │
+///                     └───────────────┘
+///                             ▲
+/// ┌───────────────┐           │            ┌───────────────┐
+/// │  Data Node 1  │◄──────────┼───────────►│  Data Node 3  │
+/// └───────────────┘           │            └───────────────┘
+///                     ┌───────┴───────┐
+///                     │   Name Node   │
+///                     └───────────────┘
+///                             ▲
+///                             │
+///                      ┌──────┴─────┐
+///                      │   Vector   │
+///                      └────────────┘
+/// ```
+///
+/// For more information, please refer to:
+///
+/// - [HDFS Users Guide](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html)
+/// - [opendal::services::hdfs](https://docs.rs/opendal/latest/opendal/services/struct.Hdfs.html)
+#[configurable_component(sink("hdfs"))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+
+pub struct HdfsConfig {
+    /// The root path of hdfs services.
+    ///
+    /// Root must be a valid dir.
+    ///
+    /// The final file path with be like `{root}/{prefix}{suffix}`.
+    #[serde(default)]
+    pub root: String,
+    /// A prefix to apply to all keys.
+    ///
+    /// Prefixes are useful for partitioning objects, such as by creating an blob key that
+    /// stores blobs under a particular "directory". If using a prefix for this purpose, it must end
+    /// in `/` to act as a directory path. A trailing `/` is **not** automatically added.
+    ///
+    /// The final file path with be like `{root}/{prefix}{suffix}`.
```

Comment:
**question**: I'm not entirely sure what the suffix is here - is that the (current) UUID + file extension?

Reply:
I imagine we'd want to enhance this down the line to have a similar controls as the [`aws_s3`](https://vector.dev/docs/reference/configuration/sinks/aws_s3/#filename_append_uuid) sink where the user has more granular control of how the files are written, but can definitely come back around to expanding config/support there.